client/src/pages/UserPage.tsx

```diff
@@ -0,0 +1,276 @@
+                        display: false,
+                      },
+                    },
+                    y: {
+                      title: {
+                        display: true,
+                        text: 'Progress',
+                      },
+                    },
+                  },
+
+                  plugins: {
+                    filler: {
+                      propagate: true,
+                    },
+                    legend: {
+                      position: 'bottom',
+                    },
+                    tooltip: {
+                      backgroundColor: 'rgb(5, 3, 29)',
+                      borderWidth: 1,
+                      cornerRadius: 6,
+                      boxHeight: 5,
+                      padding: 4,
+                      caretSize: 4,
+                      caretPadding: 2,
+                      titleFont: {
+                        size: 10,
+                        family: 'Helvetica Neue',
+                        weight: 'bold',
+                      },
+                      displayColors: false,
+                      yAlign: 'bottom',
+                      xAlign: 'left',
+                      bodySpacing: 2,
+                      titleSpacing: 2,
+                      enabled: true,
+                      mode: 'index',
+                      axis: 'y',
+                      intersect: false,
+                      position: 'nearest',
+                      bodyFont: {
+                        size: 8,
+                      },
+                      callbacks: {
+                        title: function (tooltipItems) {
+                          return tooltipItems[0].label;
+                        },
+                        label: function (context) {
+                          const dataIndex = context.dataIndex;
+                          const details = scoreHistory[dataIndex];
+                          if (
+                            details === undefined ||
+                            details.progressScore === undefined
+                          ) {
+                            return [];
+                          } else {
+                            [
+                              `Dream: ${details.dreamActivity}`,
+                              ` Depression: ${details.depressionLevel}`,
+                              `Sleep: ${details.sleepQuality}`,
+                              ` Activity: ${details.durationOfActivity}`,
+                              ` Bedtime:${details.bedtime}`,
+                            ];
+                          }
+                        },
+                      },
+                    },
+                  },
+                }}
+              />
+            </div>
+          </div>
+        </div>
+        <div className="row">
+          <div className="actions">
+            <button className="my-butt" type="submit" onClick={openModal}>
+              Progress Report
+            </button>
+            {isOpen && (
```

Comment:
Since the Modal component handles showing/hiding, you don't need the `isOpen &&` here. You can just use `<Modal>` directly.